### PR TITLE
Add model_access_config, multi_model_config and inference_specification_name to sagemaker_model primary_container and container block

### LINKED ISSUE #34795: [Enhancement]: aws_sagemaker_model resource missing ModelAccessConfig
### Terraform Core Version

1.5.7

### AWS Provider Version

5.29.0

### Affected Resource(s)

resource: aws_sagemaker_model

### Expected Behavior

I'm attempting to create an AWS Sagemaker model resource for one of the Meta Llama2 LLMs.  I've captured the model data source and needed env vars from a manually created version of the model through SageMaker Jumpstart and converted them into an aws_sagemaker_model resource.  

I would expect to be able to successfully create an instance of the model using this resource following the same pattern/configuration that JumpStart used.  

### Actual Behavior

When I attempt to create the model in my Terraform apply, I'm presented with an error indicating that model creation failed due to the model's EULA not being accepted.  

### Relevant Error/Panic Output Snippet

```shell
│ Error: creating SageMaker model: ValidationException: The AcceptEula value must be explicitly defined as True in order to accept the end-user license agreement (EULA) that this "s3://jumpstart-private-cache-prod-us-east-1/meta-textgeneration/meta-textgeneration-llama-2-13b-f/artifacts/inference-prepack/v1.0.0/" model requires. You are responsible for reviewing and complying with any applicable license terms and making sure they are acceptable for your use case before downloading or using a model.
│ 	status code: 400, request id: 4873a968-50a9-4aa5-b0b4-8af4cba6ae90
│
│   with module.use1.module.sagemaker_resources.module.models.module.llama2_13b.aws_sagemaker_model.llama2_13b,
│   on modules/sagemaker/models/llama2_13b/main.tf line 18, in resource "aws_sagemaker_model" "llama2_13b":
│   18: resource "aws_sagemaker_model" "llama2_13b" {
```

### Terraform Configuration Files

```terraform
resource "aws_sagemaker_model" "llama2_13b" {
  name = "meta-textgeneration-llama-2-13b-f-20231130-192251"
  execution_role_arn = var.model_execution_role_arn

  primary_container {
    image = "763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-tgi-inference:2.0.1-tgi1.1.0-gpu-py39-cu118-ubuntu20.04"
    mode = "SingleModel"
    environment = {
      ENDPOINT_SERVER_TIMEOUT = 3600
      HF_MODEL_ID = "/opt/ml/model"
      MAX_INPUT_LENGTH = 4095
      MAX_TOTAL_TOKENS = 4096
      MODEL_CACHE_ROOT = "/opt/ml/model"
      SAGEMAKER_ENV = 1
      SAGEMAKER_MODEL_SERVER_WORKERS = 1
      SAGEMAKER_PROGRAM = "inference.py"
      SM_NUM_GPUS = 4
      AcceptEula = "True"
      accept_eula = "true"
    }

    model_data_source {
      s3_data_source {
        compression_type = "None"
        s3_data_type     = "S3Prefix"
        s3_uri           = "s3://jumpstart-private-cache-prod-us-east-1/meta-textgeneration/meta-textgeneration-llama-2-13b-f/artifacts/inference-prepack/v1.0.0/"
      }
    }
  }

  enable_network_isolation = true

  tags = local.model_tags
}
```

### Steps to Reproduce

- Define aws_sagemaker_model resource using configuration provided above.  
- Attempt to apply Terraform to create the model resource
- Model creation fails for me usually after a few minutes

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

The AWS API endpoint for creating a SageMaker model possesses an input for `ModelAccessConfig` under `S3DataSource` which allows specification of an AcceptEula boolean value:
- https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateModel.html

The documentation for the current (5.29.0 as of creation of this issue) version of the Terraform AWS Provider doesn't indicate support for this input section.  

### References

AWS Sagemaker Create Model Endpoint Reference:
- https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateModel.html

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by deepakbshetty**
### Relations
Closes #34795
Also adds below to `primary_container` and `container`
>inference_specification_name
>multi_model_config - > model_access_config ( Enabled | Disabled )

### Output from Acceptance Testing

```
$ make testacc TESTS=TestAccSageMakerModel_primaryContainerInferenceSpecificationName PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerModel_primaryContainerInferenceSpecificationName'  -timeout 360m
=== RUN   TestAccSageMakerModel_primaryContainerInferenceSpecificationName
=== PAUSE TestAccSageMakerModel_primaryContainerInferenceSpecificationName
=== CONT  TestAccSageMakerModel_primaryContainerInferenceSpecificationName
--- PASS: TestAccSageMakerModel_primaryContainerInferenceSpecificationName (101.61s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  134.874s

$ make testacc TESTS=TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting'  -timeout 360m
=== RUN   TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
=== PAUSE TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
=== CONT  TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
--- PASS: TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting (114.44s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  147.930s

$ make testacc TESTS=TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula'  -timeout 360m
=== RUN   TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
=== PAUSE TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
=== CONT  TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
--- PASS: TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula (114.83s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  148.564s

$ make testacc TESTS=TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting'  -timeout 360m
=== RUN   TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
=== PAUSE TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
=== CONT  TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
--- PASS: TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting (110.88s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  143.968s

...
```

### CODE REVIEW

**Comment by deepakbshetty** [comment]
Hi @DrFaust92, For your kind review.

**Comment by deepakbshetty** [comment]
Hi @ewbankkit, The issues should be fixed now tests should better. Appreciate if you can retrigger. Thanks in advance.

**Comment by deepakbshetty** [comment]
Dear @DrFaust92, Gentle nudge to review

**Review by DrFaust92 — APPROVED** [review]
LGTM

File: internal/service/sagemaker/model_test.go [hunk 1]
```diff
@@ -955,3 +1068,237 @@ resource "aws_security_group" "test" {
...
+  }
+}
+`, rName))
+}
+
+func testAccModelConfig_primaryContainerMultiModelConfigModelCacheSetting(rName string) string {
+	return acctest.ConfigCompose(testAccModelConfig_base(rName), fmt.Sprintf(`
+data "aws_region" "current" {}
+data "aws_partition" "current" {}
+locals {
```
Comment (DrFaust92) [inline internal/service/sagemaker/model_test.go:1176]: consider leveraging prebuilt_ecr_image datasource

Comment (deepakbshetty) [reply internal/service/sagemaker/model_test.go:1176]: Requires a change in data source to include these new images ex. huggingface-pytorch-tgi , Hence had to fallback on url parsing with accounts

**Comment by deepakbshetty** [comment]
Hi @ewbankkit 👋 May I request if you can consider this for merge

**Comment by DrFaust92** [comment]
[deepakbshetty](https://github.com/deepakbshetty) can you fix conflicts and add changelog entry?

**Comment by deepakbshetty** [comment]
Dear @DrFaust92 , Added changelog and fixed conflicts

File: internal/service/sagemaker/sagemaker_test.go [hunk 2]
```diff
@@ -46,31 +46,49 @@ func TestAccSageMaker_serial(t *testing.T) {
 			"kernelGatewayAppSettings":                 testAccDomain_kernelGatewayAppSettings,
 			"kernelGatewayAppSettings_customImage":     testAccDomain_kernelGatewayAppSettings_customImage,
 			"kernelGatewayAppSettings_lifecycleConfig": testAccDomain_kernelGatewayAppSettings_lifecycleConfig,
-			"kernelGatewayAppSettings_defaultResourceAndCustomImage":  testAccDomain_kernelGatewayAppSettings_defaultResourceSpecAndCustomImage,
```
Comment (DrFaust92) [inline internal/service/sagemaker/sagemaker_test.go:49]: change in this file seems unrelated to this PR

**Review by DrFaust92 — CHANGES_REQUESTED** [review]

File: internal/service/sagemaker/sagemaker_test.go (hunk 2 above)
Comment (deepakbshetty) [reply internal/service/sagemaker/sagemaker_test.go]: Fixed.

File: internal/service/sagemaker/sagemaker_test.go [hunk 3]
```diff
@@ -72,6 +72,27 @@ func TestAccSageMaker_serial(t *testing.T) {
 			"posix":                                                   testAccDomain_posix,
 			"spaceStorageSettings":                                    testAccDomain_spaceStorageSettings,
 		},
+		"Model": {
```
Comment (DrFaust92) [inline internal/service/sagemaker/sagemaker_test.go:75]: whats the reason for this? this is only needed when tests need to be run serially (some resource can only be created only a single time at once). and even if that is the case here all tests function need to be lower case.

in any case, i dont this is needed

File: internal/service/sagemaker/sagemaker_test.go [hunk 4]
```diff
@@ -81,8 +102,8 @@ func TestAccSageMaker_serial(t *testing.T) {
 		},
 		"Space": {
 			acctest.CtBasic:            testAccSpace_basic,
-			acctest.CtDisappears:       testAccSpace_tags,
-			"tags":                     testAccSpace_disappears,
+			acctest.CtDisappears:       testAccSpace_disappears,
```
Comment (DrFaust92) [inline internal/service/sagemaker/sagemaker_test.go:105]: is this relevant to the PR?

Comment (deepakbshetty) [reply internal/service/sagemaker/sagemaker_test.go:105]: No, I saw a typo where functions are mismatched - tags pointing to disappears. Have reverted considering outside scope this PR

File: internal/service/sagemaker/sagemaker_test.go (hunk 3 above)
Comment (deepakbshetty) [reply internal/service/sagemaker/sagemaker_test.go:75]: Agreed. Reverted/Removed considering model constraints for serial run. There is work required to rename all old/legacy test functions if we intend to add in future.

**Review by DrFaust92 — APPROVED** [review]
LGTM (without running tests)

**Comment by deepakbshetty** [comment]
Hi 👋 @jar-b @ewbankkit , The provider Lint checks seems to have bug here. The PR has tests casing using data sources for aws region yet the checks fail. Unless I am looking  at this wrong way ?

**Comment by deepakbshetty** [comment]
Dear @DrFaust92 , Request re-review. Added prebuilt ecr image data source to tests, as they are now available in provider version 5.61.0

File: .changelog/35873.txt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+@@ -0,0 +1,7 @@
```
Comment (DrFaust92) [inline .changelog/35873.txt:1]: whats this line?

Comment (deepakbshetty) [reply .changelog/35873.txt]: Fixed now. The diff line seems to have crept in during merge conflict resolution.

**Review by DrFaust92 — APPROVED** [review]

**Comment by deepakbshetty** [comment]
Hi @ewbankkit , For your kind consideration. Adds multiple missing features for sagemaker model creation.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSageMakerModel_' PKG=sagemaker ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/sagemaker/... -v -count 1 -parallel 3  -run=TestAccSageMakerModel_ -timeout 360m
=== RUN   TestAccSageMakerModel_basic
=== PAUSE TestAccSageMakerModel_basic
=== RUN   TestAccSageMakerModel_inferenceExecution
=== PAUSE TestAccSageMakerModel_inferenceExecution
=== RUN   TestAccSageMakerModel_tags
=== PAUSE TestAccSageMakerModel_tags
=== RUN   TestAccSageMakerModel_primaryContainerModelDataURL
=== PAUSE TestAccSageMakerModel_primaryContainerModelDataURL
=== RUN   TestAccSageMakerModel_primaryContainerHostname
=== PAUSE TestAccSageMakerModel_primaryContainerHostname
=== RUN   TestAccSageMakerModel_primaryContainerImage
=== PAUSE TestAccSageMakerModel_primaryContainerImage
=== RUN   TestAccSageMakerModel_primaryContainerEnvironment
=== PAUSE TestAccSageMakerModel_primaryContainerEnvironment
=== RUN   TestAccSageMakerModel_primaryContainerModeSingle
=== PAUSE TestAccSageMakerModel_primaryContainerModeSingle
=== RUN   TestAccSageMakerModel_primaryContainerModelPackageName
=== PAUSE TestAccSageMakerModel_primaryContainerModelPackageName
=== RUN   TestAccSageMakerModel_primaryContainerModelDataSource
=== PAUSE TestAccSageMakerModel_primaryContainerModelDataSource
=== RUN   TestAccSageMakerModel_containers
=== PAUSE TestAccSageMakerModel_containers
=== RUN   TestAccSageMakerModel_vpc
=== PAUSE TestAccSageMakerModel_vpc
=== RUN   TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
=== PAUSE TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
=== RUN   TestAccSageMakerModel_networkIsolation
=== PAUSE TestAccSageMakerModel_networkIsolation
=== RUN   TestAccSageMakerModel_disappears
=== PAUSE TestAccSageMakerModel_disappears
=== RUN   TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
=== PAUSE TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
=== RUN   TestAccSageMakerModel_primaryContainerInferenceSpecificationName
=== PAUSE TestAccSageMakerModel_primaryContainerInferenceSpecificationName
=== RUN   TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
=== PAUSE TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
=== RUN   TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
=== PAUSE TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
=== CONT  TestAccSageMakerModel_basic
=== CONT  TestAccSageMakerModel_containers
=== CONT  TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula
--- PASS: TestAccSageMakerModel_containers (22.84s)
=== CONT  TestAccSageMakerModel_primaryContainerImage
--- PASS: TestAccSageMakerModel_primaryContainerModelS3DataSourceAcceptEula (22.93s)
=== CONT  TestAccSageMakerModel_primaryContainerModelDataSource
--- PASS: TestAccSageMakerModel_basic (27.84s)
=== CONT  TestAccSageMakerModel_primaryContainerModelPackageName
--- PASS: TestAccSageMakerModel_primaryContainerModelDataSource (25.78s)
=== CONT  TestAccSageMakerModel_primaryContainerModeSingle
--- PASS: TestAccSageMakerModel_primaryContainerModelPackageName (22.54s)
=== CONT  TestAccSageMakerModel_primaryContainerEnvironment
--- PASS: TestAccSageMakerModel_primaryContainerImage (27.78s)
=== CONT  TestAccSageMakerModel_primaryContainerModelDataURL
--- PASS: TestAccSageMakerModel_primaryContainerEnvironment (21.20s)
=== CONT  TestAccSageMakerModel_primaryContainerHostname
--- PASS: TestAccSageMakerModel_primaryContainerModeSingle (26.26s)
=== CONT  TestAccSageMakerModel_networkIsolation
--- PASS: TestAccSageMakerModel_primaryContainerModelDataURL (27.18s)
=== CONT  TestAccSageMakerModel_disappears
--- PASS: TestAccSageMakerModel_disappears (20.28s)
=== CONT  TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting
--- PASS: TestAccSageMakerModel_networkIsolation (25.28s)
=== CONT  TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting
--- PASS: TestAccSageMakerModel_primaryContainerHostname (32.05s)
=== CONT  TestAccSageMakerModel_primaryContainerInferenceSpecificationName
--- PASS: TestAccSageMakerModel_primaryContainerMultiModelConfigModelCacheSetting (24.70s)
=== CONT  TestAccSageMakerModel_tags
--- PASS: TestAccSageMakerModel_containersMultiModelConfigModelCacheSetting (25.98s)
=== CONT  TestAccSageMakerModel_inferenceExecution
--- PASS: TestAccSageMakerModel_primaryContainerInferenceSpecificationName (26.21s)
=== CONT  TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
--- PASS: TestAccSageMakerModel_inferenceExecution (22.70s)
=== CONT  TestAccSageMakerModel_vpc
--- PASS: TestAccSageMakerModel_primaryContainerPrivateDockerRegistry (27.31s)
--- PASS: TestAccSageMakerModel_tags (41.34s)
--- PASS: TestAccSageMakerModel_vpc (26.29s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker	180.223s
```

**Comment by ewbankkit** [comment]
@deepakbshetty Thanks for the contribution :tada: :clap:.
